extensions/rv32im/circuit/src/adapters/mod.rs

```diff
@@ -52,101 +52,118 @@ pub fn decompose<F: PrimeField32>(value: u32) -> [F; RV32_REGISTER_NUM_LIMBS] {
 }
 
 /// Atomic read operation which increments the timestamp by 1.
-/// Returns `(t_prev, [reg_ptr:4]_1)` where `t_prev` is the timestamp of the last memory access.
+/// Returns `(t_prev, [ptr:4]_{address_space})` where `t_prev` is the timestamp of the last memory access.
 #[inline(always)]
-pub fn timed_read_reg(
+pub fn timed_read(
     memory: &mut TracingMemory,
-    reg_ptr: u32,
+    address_space: u32,
```

Comment:
I actually find it annoying to have to specify the address space, and also I'm not sure if there's any runtime impact of using a variable instead of a constant -- for both reasons I'd prefer if we still kept `_read_reg` vs `_read_mem` versions

Reply:
okay, i can split it into two functions. there'll probably also be a version for reading from the hint space but i'm not sure if the alignment guarantees hold for that